grammar.js

```diff
@@ -30,42 +32,67 @@ module.exports = grammar({
   name: "git_commit",
 
   extras: ($) => [WHITE_SPACE],
-
   rules: {
     source: ($) =>
       seq(
-        optional(choice($.comment, $.subject)),
-        optional(seq(NEWLINE, repeat($._body_line))),
+        optional(choice($.subject, $.comment)),
+        repeat($._body_line),
+        repeat($._trailer),
+        optional(repeat1($._newline)),
         optional(
```

Comment:
important refactor! I started each of `$._body_line` and `$._trailer` with a `$._newline` to avoid marking trailing newlines as optional/repeated.